### PR TITLE
lvgl/contrib: allow for SDL display driver height/width to be adjusted

### PR DESCRIPTION
**Pull request by tvanfossen-bissell**
### Contribution description

When running a RIOT native app using SDL, the display window for SDL is adjustable, but the display driver used by LVGL is not - it always defaults to the 320x240 size provided in the lv_disp_drv_init.

When driving an LCD, the resolution for the disp driver is updated post-init of the driver. Following the similar conventions, this tidbit references LCD_SCREEN_HEIGHT/WIDTH defines, which can be passed to an app as a CFLAG in a Makefile.include, or defaulting to the SDL_HOR/VER_RES defines passed in through lv_drv_conf.h in lv_drivers package.

This enables SDL native applications to better emulate the exact resolution of an LCD screen.

### Testing procedure

Using a test package for a custom LVGL RIOT application, I verified that SDL_HOR/VER_RES can adjust the SDL window, but not the display driver used by LVGL when driving an SDL window.

No other defines provided by lv_conf.h, lv_drv_conf.h, or lvgl_riot_conf.h adjust the size of the SDL emulated display driven by LVGL. The simplest fix to me was to adjust the size of the display driver post-init as is done for the non-SDL display driver. This fix enables the emulated display size to be adjusted

### Issues/PRs references

Resolves issue #18461

### CODE REVIEW

File: pkg/lvgl/contrib/lvgl.c [hunk 1]
```diff
@@ -193,5 +208,6 @@ void lvgl_run(void)
 void lvgl_wakeup(void)
 {
     thread_t *tcb = thread_get(_task_thread_pid);
+
```
Comment (maribu) [inline pkg/lvgl/contrib/lvgl.c:211]:
```suggestion
```

This addition of white space seems unrelated

**Review by maribu — APPROVED** [review]
ACK. Code change looks good to me. I sadly have a 64 bit only distro and running SDL from within docker is not trivial, so I cannot easily test this. But I guess I can just trust your testing.

If you agree to drop the addition of the white space at an unrelated function, please squash right in. Otherwise I'd merge as is.

**Comment by tvanfossen-bissell** [comment]
> ACK. Code change looks good to me. I sadly have a 64 bit only distro and running SDL from within docker is not trivial, so I cannot easily test this. But I guess I can just trust your testing.
> 
> If you agree to drop the addition of the white space at an unrelated function, please squash right in. Otherwise I'd merge as is.

Updated, was not intentional

image
with flags:
CFLAGS += -DSDL_HOR_RES=300
CFLAGS += -DSDL_VER_RES=400
CFLAGS += -DLCD_SCREEN_WIDTH=240
CFLAGS += -DLCD_SCREEN_HEIGHT=320

image
with flags
with flags:
CFLAGS += -DSDL_HOR_RES=300
CFLAGS += -DSDL_VER_RES=400
CFLAGS += -DLCD_SCREEN_WIDTH=320 //runs off sdl screen
CFLAGS += -DLCD_SCREEN_HEIGHT=240

**Comment by maribu** [comment]
Would you mind squashing the two commits into one?

You can do so e.g. using

```
git rebase -i HEAD~2
```

and then replace `pick` by `fixup` in front of the `Update: ...` commit.

**Comment by tvanfossen-bissell** [comment]
fixed - thanks for the pointer

**Review by maribu — APPROVED** [review]

File: pkg/lvgl/contrib/lvgl.c [hunk 2]
```diff
@@ -141,6 +154,8 @@ void lvgl_init(screen_dev_t *screen_dev)
     sdl_init();
     /* Used when `LV_VDB_SIZE != 0` in lv_conf.h (buffered drawing) */
     disp_drv.flush_cb = sdl_display_flush;
+    disp_drv.hor_res = LCD_SCREEN_WIDTH
+    disp_drv.ver_res = LCD_SCREEN_HEIGHT
```
Comment (maribu) [inline pkg/lvgl/contrib/lvgl.c:158]:
```suggestion
    disp_drv.ver_res = LCD_SCREEN_HEIGHT;
```

File: pkg/lvgl/contrib/lvgl.c [hunk 3]
```diff
@@ -141,6 +154,8 @@ void lvgl_init(screen_dev_t *screen_dev)
     sdl_init();
     /* Used when `LV_VDB_SIZE != 0` in lv_conf.h (buffered drawing) */
     disp_drv.flush_cb = sdl_display_flush;
+    disp_drv.hor_res = LCD_SCREEN_WIDTH
```
Comment (maribu) [inline pkg/lvgl/contrib/lvgl.c:157]:
```suggestion
    disp_drv.hor_res = LCD_SCREEN_WIDTH;
```

**Review by maribu — COMMENTED** [review]
Looks like something went wrong during squashing. You could add the semicolons locally and commit with `--amend` to directly add this into the current commit. A `git push --force-with-lease` is again needed to push the changed history into the PR.

**Comment by maribu** [comment]
Thx for your contribution and congratulations for your first PR merged :)

I almost forget this during the excitement of the annual [RIOT summit](https://summit.riot-os.org/)
